### ISSUE TITLE
Feature Request : NVR Tiered Storage

### ISSUE THREAD

**Issue by briangunderson** [koush/nvr.scrypted.app#4]
It would be nice to be able to have Scrypted NVR be able to use multiple tiers of storage locations. Blue Iris does this fairly well. Using tiered storage, NVR would save new clips to fast storage such as SSD, and when clips need to age out to make room for newer clips, rather than being deleted they are instead moved to secondary/tertiary disks which most often will be spinning disks. Indexes, snapshots, etc are ideally kept in primary storage along with the newest clips, this obviously helps with responsiveness. Thanks for your consideration!

**Comment by koush** [comment]
I was thinking about tiering storage by stream type. the low res stream needs to be on fast storage for timeline scrub. the high res stream can be on spinning disks. currently it cleans out high res streams before low res streams to retain space. but this is the opposite of what you want, as it is optimized for timeline responsiveness. need to think about this some more.

**Comment by briangunderson** [comment]
My thinking is that I would want super responsive high performance
scrubbing to review events that took place recently (last hour? day?) and
for events that took place a long time ago (weeks?) I would be more
tolerant of a slow disk. That's just my opinion and what I got used to with
blue iris. It might not be the best architecture but it makes sense to me.

**Comment by koush** [comment]
scrypted scrubbing uses the low resolution stream. the contributors for scrub performance is the availability of said substream, and fast seek times on disk.

are you experiencing issues scrubbing right now?

**Comment by briangunderson** [comment]
Scrypted is light years ahead of anything else I've tried. It is FAST. That
said, when I migrated from Blue Iris I didn't swap out my 3-tier storage
solution for something more suitable for Scrypted's storage architecture,
so I'm "stuck" using just my big slow 3.5" drives and forgoing the NVMe SSD
and 15k SAS drives that previously served as my "fastest" and "fast" tiers.
I'm a network engineer by trade and not a storage expert so it's quite
possible that the architecture I'm envisioning isn't actually the
architecture that would yield the best results. You seem to have a pretty
good handle on tuning real-time digital video systems! :)
